.github/scripts/test_trymerge.py

```diff
@@ -806,13 +808,92 @@ def test_dont_ignore_flaky_failures(self, *args: Any) -> None:
+@mock.patch("trymerge.gh_graphql", side_effect=mocked_gh_graphql)
+@mock.patch("trymerge.gh_fetch_merge_base", return_value="")
+@mock.patch("trymerge.get_drci_classifications", return_value={})
+class TestBypassFailuresOnSandCastle(TestCase):
+    def test_get_classifications(self, *args: Any) -> None:
+        pr = GitHubPR("pytorch", "pytorch", 111467)
+        checks = pr.get_checkrun_conclusions()
+        checks = get_classifications(
+            pr.pr_num,
+            pr.project,
+            checks,
+            [],
+        )
+        pending, failed, ignorable = categorize_checks(checks, list(checks.keys()))
+        self.assertTrue(len(pending) == 0)
+        self.assertTrue(len(failed) == 0)
+        self.assertTrue(len(ignorable["FLAKY"]) == 1)
+        self.assertTrue(len(ignorable["BROKEN_TRUNK"]) == 1)
+
+    def test_get_classifications_drci_checkrun_not_found(self, *args: Any) -> None:
+        pr = GitHubPR("pytorch", "pytorch", 111467)
+
+        # No summary
+        checks = pr.get_checkrun_conclusions()
+        checks[DRCI_CHECKRUN_NAME] = JobCheckState(
+            DRCI_CHECKRUN_NAME,
+            "",
+            "NEUTRAL",
+            None,
+            1,
+            "",
+            None,
+        )
+        checks = get_classifications(
+            pr.pr_num,
+            pr.project,
+            checks,
+            [],
+        )
+        pending, failed, ignorable = categorize_checks(checks, list(checks.keys()))
+        self.assertTrue(len(pending) == 0)
+        self.assertTrue(len(failed) == 2)
+
+        # Empty summary
+        checks = pr.get_checkrun_conclusions()
+        checks[DRCI_CHECKRUN_NAME] = JobCheckState(
+            DRCI_CHECKRUN_NAME,
+            "",
+            "NEUTRAL",
+            None,
+            1,
+            "",
+            "",
+        )
+        checks = get_classifications(
+            pr.pr_num,
+            pr.project,
+            checks,
+            [],
+        )
+        pending, failed, ignorable = categorize_checks(checks, list(checks.keys()))
+        self.assertTrue(len(pending) == 0)
+        self.assertTrue(len(failed) == 2)
+
+        # No Dr.CI checkrun
+        checks = pr.get_checkrun_conclusions()
+        del checks[DRCI_CHECKRUN_NAME]
+        checks = get_classifications(
+            pr.pr_num,
+            pr.project,
+            checks,
+            [],
+        )
+        pending, failed, ignorable = categorize_checks(checks, list(checks.keys()))
+        self.assertTrue(len(pending) == 0)
+        self.assertTrue(len(failed) == 2)
+
+
 @mock.patch("trymerge.get_rockset_results", side_effect=mocked_rockset_results)
 @mock.patch("trymerge.gh_graphql", side_effect=mocked_gh_graphql)
 @mock.patch("trymerge.gh_fetch_merge_base", return_value="")
 @mock.patch(
     "trymerge.get_drci_classifications", side_effect=mocked_drci_classifications
 )
-class TestGitHubPRGhstackDependencies2(TestCase):
+class TestGitHubPRGhstackDependencies(TestCase):
```

Comment:
Also unrelated. I don't know why we call this `TestGitHubPRGhstackDependencies2` lol